bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -115,30 +116,42 @@ public class DropTarget extends Widget {
 	static final String DEFAULT_DROP_TARGET_EFFECT = "DEFAULT_DROP_TARGET_EFFECT"; //$NON-NLS-1$
 	static final String IS_ACTIVE = "org.eclipse.swt.internal.isActive"; //$NON-NLS-1$
 
+	public void handleLabels(Control c, long cls) {
+		// If the content view can be an image view, then add the dragging methods to the image view too.
+	    // This is used by Label so that dragging can work even when the Label has an image set on it.
+	    if (!labelDragHandlersAdded) {
+	        return;
+	    }
+	    if (c instanceof Label) {
+	        addDragHandlers(cls);  // adding the handlers to label class
```

Comment:
Is this the cls of the Label? I see that cls is not being recomputed for a control.

Reply:
@lshanmug Could you brief on the requested change?

Reply:
If the initial `cls` passed-in is not being used, it can even be removed as input param for `handleLabels` and just be computed at the point of use (line 126).